crates/router/src/core/payments/operations/payment_create.rs

```diff
@@ -846,13 +860,11 @@ impl PaymentCreate {
 
         let mut additional_pm_data = request
             .payment_method_data
-            .as_ref()
-            .and_then(|payment_method_data_request| {
-                payment_method_data_request.payment_method_data.as_ref()
-            })
-            .async_map(|payment_method_data| async {
+            .clone()
+            .and_then(|payment_method_data_request| payment_method_data_request.payment_method_data)
+            .async_and_then(|payment_method_data| async {
```

Comment:
No need to clone entire payment_method_data of request. Instead take `request.payment_method_data` as reference and clone `payment_method_data_request.payment_method_data` inside and_then